integration-tests/tests/examples_test.go

```diff
@@ -173,3 +179,96 @@ func (s *configExampleSuite) TestPrintMessageFromConfig(c *check.C) {
 		c.Assert(output, check.Equals, t.message, check.Commentf("Wrong message"))
 	}
 }
+
+var _ = check.Suite(&licensedExampleSuite{})
+
+type licensedExampleSuite struct {
+	common.SnappySuite
+}
+
+func (s *licensedExampleSuite) TestAcceptLicenseMustInstallSnap(c *check.C) {
+	cmd := exec.Command("sudo", "snappy", "install", "licensed.canonical")
+	f, err := pty.Start(cmd)
+	c.Assert(err, check.IsNil, check.Commentf("Error starting pty: %s", err))
+	defer common.RemoveSnap(c, "licensed.canonical")
+
+	s.assertLicense(c, f)
+	// Accept the license.
+	_, err = f.Write([]byte("y\n"))
+	c.Assert(err, check.IsNil, check.Commentf("Error writing to pty: %s", err))
+
+	cmd.Wait()
+
+	c.Assert(s.isSnapInstalled(c), check.Equals, true, check.Commentf("The snap was not installed"))
+}
+
+func (s *licensedExampleSuite) TestDeclineLicenseMustNotInstallSnap(c *check.C) {
+	cmd := exec.Command("sudo", "snappy", "install", "licensed.canonical")
+	f, err := pty.Start(cmd)
+	c.Assert(err, check.IsNil, check.Commentf("Error starting pty: %s", err))
+
+	s.assertLicense(c, f)
+	// Decline the license.
+	_, err = f.Write([]byte("n\n"))
+	c.Assert(err, check.IsNil, check.Commentf("Error writing to pty: %s", err))
+
+	cmd.Wait()
+
+	c.Assert(s.isSnapInstalled(c), check.Equals, false, check.Commentf("The snap was installed"))
+}
+
+func (s *licensedExampleSuite) assertLicense(c *check.C, f *os.File) {
+	output := s.readUntilPrompt(c, f)
+	expected := "Installing licensed.canonical" +
+		"Starting download of licensed" +
+		".*" +
+		"Done" +
+		"licensed requires that you accept the following license before continuing" +
+		"This product is meant for educational purposes only. .* No other warranty expressed or implied."
+	c.Assert(output, check.Matches, expected)
+}
+
+func (s *licensedExampleSuite) readUntilPrompt(c *check.C, f *os.File) string {
+	var output string
+	scanner := bufio.NewScanner(f)
+
+	scanLinesUntilPrompt := func(data []byte, atEOF bool) (advance int, token []byte, err error) {
+		if atEOF && len(data) == 0 {
```

Comment:
I copied this from the ScanLines function, just hacked it to return earlier instead of getting stuck waiting for a \n. It seems that by the time they enter this if, the scanner.error property is already set. That's the only way to get atEOF = true. So the loop is exiting anyway. If they return an error here, the only thing that changes is that scanner.start and scanner.end are not set back to 0.